lib/vector-core/src/event/metric.rs

```diff
@@ -726,6 +779,24 @@ impl MetricData {
         }
     }
 
+    /// Creates a new `MetricData` from individual parts.
+    pub fn from_parts(
+        timestamp: Option<DateTime<Utc>>,
+        kind: MetricKind,
+        value: MetricValue,
+    ) -> Self {
+        Self {
+            timestamp,
+            kind,
+            value,
+        }
+    }
+
+    /// Consumes this `MetricData` and returns its individual parts.
+    pub fn into_parts(self) -> (Option<DateTime<Utc>>, MetricKind, MetricValue) {
+        (self.timestamp, self.kind, self.value)
+    }
+
```

Comment:
Necessary? No.  This is my personal bias coming through: I strongly dislike the structure destructing pattern because `cargo fmt` usually ends up stratifying it across multiple lines, and renaming fields can become very verbose.  Dedicated `from_*`/`into_*` methods allow for a more succinct bit of code.

Now that I come to think of it, I also dislike `pub` fields because it makes encapsulation harder..

I could peruse the codebase to see if there's a slant we typically take -- structure destructing vs `from_*`/`into_*` -- since, if anything, I should really be going off our existing style.

Reply:
On the other hand, destructuring retains the names while `from_parts`/`into_parts` only has argument/tuple positioning (just like the tuple structs I mentioned elsewhere). I do agree about the encapsulation issue, so 🤷🏻

Reply:
Proposal: I leave the above changes as-is, _but_ I create a new issue for defining an RFC where we adopt a style guide, which would codify patterns for scenarios like these that we want to enforce.

I'm not necessarily against the style you propose, and would be happy to propose it as a default in the hypothetical RFC/style guide, but it seems worth codifying/agreeing upon it officially if we care enough to bring it up in code reviews.

Does this seem like an amenable resolution of this comment, in your eyes?

Reply:
@bruceg Proposal: I leave the above changes as-is, _but_ I create a new issue for defining an RFC where we adopt a style guide, which would codify patterns for scenarios like these that we want to enforce.

I'm not necessarily against the style you propose, and would be happy to propose it as a default in the hypothetical RFC/style guide, but it seems worth codifying/agreeing upon it officially if we care enough to bring it up in code reviews.

Does this seem like an amenable resolution of this comment, in your eyes?